### PR TITLE
feat(datepicker): add uib- prefix

### PR DESCRIPTION
**Pull request by wesleycho**
Might need assistance here, but here's what I have - I refactored the linking functions into controllers so the code can be reused between the deprecated and new prefixed directives.

### CODE REVIEW

**Comment by wesleycho** [comment]
This PR is good for review now.

File: src/datepicker/datepicker.js [hunk 1]
```diff
@@ -492,332 +519,489 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
...
-      scope.showButtonBar = angular.isDefined(attrs.showButtonBar) ? scope.$parent.$eval(attrs.showButtonBar) : datepickerPopupConfig.showButtonBar;
-
-      scope.getText = function(key) {
-        return scope[key + 'Text'] || datepickerPopupConfig[key + 'Text'];
-      };
+.controller('UibDatepickerPopupController', ['$scope', '$element', '$attrs', '$compile', '$parse', '$document', '$rootScope', '$uibPosition', 'dateFilter', 'uibDateParser', 'uibDatepickerPopupConfig', '$timeout',
+function(scope, element, attrs, $compile, $parse, $document, $rootScope, $position, dateFilter, dateParser, datepickerPopupConfig, $timeout) {
+  var self = this;
+  var cache = {},
+    isHtml5DateInput = false;
```
Comment (BobbieBarker) [inline src/datepicker/datepicker.js]: is this missing a var for declaration?

**Comment by llafuente** [comment]
This is a breaking change: `datepickerConfig` and `datepickerPopupConfig` no longer exists.

**Comment by icfantv** [comment]
@llafuente, good catch.  Thanks.

**Pull request closed without merging**